plan/resolve_indices.go

```diff
@@ -245,6 +245,29 @@ func (p *Insert) ResolveIndices() {
 		set.Col.ResolveIndices(p.tableSchema)
 		set.Expr.ResolveIndices(p.tableSchema)
 	}
+	for _, expr := range p.GenCols.Exprs {
+		expr.ResolveIndices(p.tableSchema)
+	}
+	for _, asgn := range p.GenCols.OnDuplicates {
+		asgn.Col.ResolveIndices(p.tableSchema)
+		asgn.Expr.ResolveIndices(p.tableSchema)
+	}
+}
+
+// ResolveIndices implements Plan interface.
+func (p *DataSource) ResolveIndices() {
+	p.basePlan.ResolveIndices()
+	for _, expr := range p.GenValues {
+		expr.ResolveIndices(p.Schema())
+	}
+}
+
+// ResolveIndices implements Plan interface.
+func (p *PhysicalTableScan) ResolveIndices() {
+	p.basePlan.ResolveIndices()
+	for _, expr := range p.GenValues {
+		expr.ResolveIndices(p.Schema())
+	}
 }
```

Comment:
If a virtual generated column has index, for example:
```sql
create table x (
    a int,
    b int,
    c int as (a+b) virtual,
    key (c)
);
```
the columns `c` is calculated and stored in index, so we don't need replace it by generation expression. So we don't need additional resolve_indices.